code/graphics/opengl/gropengltnl.cpp

```diff
@@ -308,8 +308,9 @@ void* gr_opengl_map_buffer(gr_buffer_handle handle)
 	Assertion(GLAD_GL_ARB_buffer_storage != 0, "Persistent mapping is not available in this OpenGL context!");
 
 	opengl_bind_buffer_object(handle);
+	glBufferData(buffer_obj.type, buffer_obj.size, nullptr, buffer_obj.gl_usage);
 	return glMapBufferRange(buffer_obj.type, 0, buffer_obj.size,
-	                        GL_MAP_WRITE_BIT | GL_MAP_PERSISTENT_BIT | GL_MAP_FLUSH_EXPLICIT_BIT);
+	                        GL_MAP_WRITE_BIT | GL_MAP_PERSISTENT_BIT | GL_MAP_FLUSH_EXPLICIT_BIT | GL_MAP_INVALIDATE_BUFFER_BIT);
```

Comment:
This call has a completely different use case than what is described in the article to which you linked. Here we are mapping the buffer only once when it has been created and then we reuse the pointer for updating the data.